### PR TITLE
[Keyboard] Add Chocofly v1

### PR DESCRIPTION
**Pull request by winder**
## Description

Add support for the open source chocofly keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/chocofly/v1/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    [_RAISE] = LAYOUT(
+                 KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_CIRC,                                     KC_AMPR, KC_ASTR, KC_MINS, KC_UNDS, KC_PLUS, KC_EQL,
+                 _______, XXXXXXX, XXXXXXX, KC_LBRC, KC_RBRC, XXXXXXX,                                     XXXXXXX, KC_EXLM, KC_AT,   KC_HASH, XXXXXXX, XXXXXXX,
+                 _______, XXXXXXX, XXXXXXX, KC_LPRN, KC_RPRN, KC_DEL,  _______,                            KC_BSPC, KC_DLR,  KC_PERC, KC_CIRC, XXXXXXX, XXXXXXX,
+                 _______, XXXXXXX, XXXXXXX, KC_LCBR, KC_RCBR, XXXXXXX, _______, _______, _______, _______, XXXXXXX, KC_AMPR, KC_ASTR, KC_PCMM, XXXXXXX, _______,
+                                                     _______, _______, _______, _______, _______, _______, _______, _______
+    ),
+};
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
```
Comment (drashna) [inline keyboards/chocofly/v1/keymaps/default/keymap.c:45]: Just a heads up, you may want to check out the new encoder map sub-feature: 
https://docs.qmk.fm/#/feature_encoders?id=encoder-map

Entirely optional, but figured it was worth mentioning.

File: keyboards/chocofly/v1/v1.h [hunk 2]
```diff
@@ -0,0 +1,28 @@
+// Copyright 2022 Vitaly Volkov (@vlkv)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "../chocofly.h"
```
Comment (drashna) [inline keyboards/chocofly/v1/v1.h:6]:
```suggestion
#include "chocofly.h"
```

File: keyboards/chocofly/v1/v1.c [hunk 3]
```diff
@@ -0,0 +1,4 @@
+// Copyright 2022 Vitaly Volkov (@vlkv)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "chocofly.h"
```
Comment (drashna) [inline keyboards/chocofly/v1/v1.c:4]:
```suggestion
#include "v1.h"
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/chocofly/chocofly.c [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+#include "chocofly.h"
+
+#ifdef ENCODER_ENABLE
+bool encoder_update_kb(uint8_t index, bool clockwise) {
+    if (!encoder_update_user(index, clockwise)) { return false; }
+    if (clockwise) {
+        tap_code(KC__VOLUP);
+    } else {
+        tap_code(KC__VOLDOWN);
```
Comment (fauxpark) [inline keyboards/chocofly/chocofly.c:12]:
```suggestion
        tap_code(KC_VOLU);
    } else {
        tap_code(KC_VOLD);
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/chocofly/v1/keymaps/winder/keymap.c [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    ),
+};
+
+#ifdef ENCODER_ENABLE
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    if (index == 0) {
+        if (clockwise) {
+            tap_code(KC__VOLDOWN);
+        } else {
+            tap_code(KC__VOLUP);
```
Comment (winder) [inline keyboards/chocofly/v1/keymaps/winder/keymap.c:49]:
```suggestion
            tap_code(KC_VOLD);
        } else {
            tap_code(KC_VOLU);
```

File: keyboards/chocofly/chocofly.c (hunk 4 above)
Comment (winder) [reply keyboards/chocofly/chocofly.c]: Thank you, I converted these to the single underscore version.

File: keyboards/chocofly/readme.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
+# Chocofly
+
+![Chocofly v1.0](https://github.com/vlkv/chocofly/blob/master/images/chocofly_v1-0.jpg)
+
+An open source ergonomic monoblock 60% keyboard.
+
+* Keyboard Maintainer: [Vitaly Volkov](https://github.com/vlkv)
+* Hardware Supported: Chocofly PCB v1, Pro Micro 5V/16MHz and compatible.
+* Hardware Availability: [Avalanche](https://github.com/vlkv/chocofly).
```
Comment (noroadsleft) [inline keyboards/chocofly/readme.md:9]:
```suggestion
* Hardware Availability: [github.com/vlkv/chocofly](https://github.com/vlkv/chocofly)
```

File: keyboards/chocofly/v1/keymaps/winder/keymap.c [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+#define FN_2 MO(_RAISE)
+#define TRMINAL LGUI(KC_ENT)
+#define RESIZE  LGUI(KC_R)
+#define BROWSER LSG(KC_ENT) // Left Shift + GUI, ENT.
+
+// required for my PC
+#undef KC_VOLU
+#undef KC_VOLD
+#define KC_VOLU KC__VOLUP
+#define KC_VOLD KC__VOLDOWN
```
Comment (noroadsleft) [inline keyboards/chocofly/v1/keymaps/winder/keymap.c:22]: Strange that the standard keycode wouldn't work. :confused:

**Review by noroadsleft — APPROVED** [review]
Looks good to me.

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**